### PR TITLE
Hierarchical tags, better tags management

### PR DESCRIPTION
**Pull request by darkside**
Adds tree-like structure to tags and the ability to manage both flat and hierarchical tags.

### CODE REVIEW

File: app/helpers/pulitzer/main_helper.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+module Pulitzer
+  module MainHelper
+    # Will this clash with helper methods like devise, cancan?
+    def authorized?
```
Comment (darkside) [inline app/helpers/pulitzer/main_helper.rb]: @gomayonqui we need to test this with the InvitedHome app

Comment (gomayonqui) [inline app/helpers/pulitzer/main_helper.rb]: should't we change this method name in case it clashes with another one from devise or something similar?

Comment (darkside) [inline app/helpers/pulitzer/main_helper.rb]: good point, I forgot to test this.

**Comment by edraut** [comment]
👍 Looks great!

**Pull request merged**